client/docs/modules.md

```diff
@@ -0,0 +1,39 @@
+# Monorepo: Modules
+
+Modules are the packages that can be found in the `modules/` folder. They're intended for pure ESM output, and are compiled rapidly using Estrella, an esbuild-based utility. They have an important file structure (don't change it), which looks like the following:
+
+```
+module:
+- src/
+  - index.ts
+  - **/*.ts
+- tests/
+  - *.ts
+- package.json
+- tsconfig.json
+```
+
+The `src/` folder is compiled into a `dist/` folder when the module is built. Similarly, files in the `tests/` folder are compiled into the `tests/dist/` folder when the tests are built.
+
+Other than the inheritance of development dependencies and tooling configurations, these modules act like fairly normal NPM packages. If you need to add a dependency to one, you can either navigate to the module and do a normal `npm i <dep>` command, or manually add it to the `package.json`. After you do that, you need to make sure to run the `npm run bootstrap` command at the root. This ensures that all the fragile symlinks get restored by Lerna.
```

Comment:
Do you mean `npm i --save <dep>`?

Reply:
`npm i <dep>` automatically adds the package as a dependency. If it didn't, that would be kinda weird.

Reply:
It does? That is not my understanding of how `npm install` works.

Okay then.

Reply:
Just to explain why it would be weird - NPM tries to keep your `node_modules` folder clean and deduped, and minimizes anything that _isn't_ your dependencies. So, if you install a package, didn't save it, then installed another one, the first one would get deleted. So, it's assumed installing a package means adding it as a dependency, which it is 99% of the time.